src/utils/createTabList.ts

```diff
@@ -0,0 +1,27 @@
+import { batch, createSignal } from 'solid-js';
+import { Tab } from '..';
+
+export const createTabList = (initialTabs: Tab[]): [() => Tab[], (t: Tab[]) => void] => {
+  let sourceSignals: { [key: string]: [get: () => string, set: (v: string) => string] } = {};
```

Comment:
Usage of [`Record`](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype) is probably better and clearer in that context 
```suggestion
  let sourceSignals: Record<string, [get: () => string, set: (value: string) => string]> = {};
```